src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/pipeline.js

```diff
@@ -108,24 +107,44 @@ export default class Pipeline {
     return result;
   }
 
-  // Updates the state of the pipeline and processors with the results
-  // from an ingest simulate call.
-  applySimulateResults(results) {
-    //update the outputObject of each processor
-    results.forEach((result) => {
+  _updateProcessorOutputs(simulateResults) {
+    simulateResults.forEach((result) => {
       const processor = this.getProcessorById(result.processorId);
 
       processor.outputObject = _.get(result, 'output');
       processor.error = _.get(result, 'error');
     });
+  }
 
-    //update the inputObject of each processor
-    results.forEach((result) => {
-      const processor = this.getProcessorById(result.processorId);
+  //Updates the error state of the pipeline and its processors
+  //If a pipeline compile error is returned, lock all processors but the error
+  //If a pipeline data error is returned, lock all processors after the error
+  _updateErrorState() {
+    this.hasCompileError = _.some(this.processors, (processor) => {
+      return _.get(processor, 'error.compile');
+    });
+    _.forEach(this.processors, processor => {
+      processor.locked = false;
+    });
 
+    const errorIndex = _.findIndex(this.processors, 'error');
+    if (errorIndex === -1) return;
+
+    _.forEach(this.processors, (processor, index) => {
+      if (this.hasCompileError && index !== errorIndex) {
+        processor.locked = true;
```

Comment:
> There is only one processor that is in error at any given time.

That's not true. The `findIndex` expression finds the first error, whether it's a compilation error or not. So let's say you have two processors, A and B, where A is a parent of B. B has a compilation error, A has either type of error. B gets locked because it's not the first error in the pipeline and the user is now stuck because they can only edit A.

I can think of two situations where this would happen:

1. The user edits B and then quickly edits A before the server response returns. This is actually quite likely if the network is slow.

2. The user edits A then B, but the response for B comes back first. Less likely, but still possible.

Since a compilation error blocks everything else, I think a processor that has been reported with a compilation error should not be locked until that compilation error has been fixed.

Here's a screenshot of scenario 1 for proof:

<img width="563" alt="screen shot 2016-03-29 at 10 14 35 am" src="https://cloud.githubusercontent.com/assets/6239176/14112077/4a418bb6-f59b-11e5-9e8e-67a73af8a66d.png">